common/workunit/workunit.cpp

```diff
@@ -2209,14 +2209,14 @@ class CSecureConstWUIterator : public CInterfaceOf<IConstWorkUnitIterator>
     bool checkScope(const char *scopeName)
     {
         int *perms = scopePermissions.getValue(scopeName);
-        int perm;
+        SecAccessFlags perm;
         if (!perms)
         {
-            perm = secuser.get() ? secmgr->authorizeWorkunitScope(*secuser, scopeName):-1;
+            perm = secuser.get() ? secmgr->authorizeWorkunitScope(*secuser, scopeName) : SecAccess_Unavailable;
             scopePermissions.setValue(scopeName, perm);
         }
         else
-            perm = *perms;
+            perm = (SecAccessFlags)*perms;
```

Comment:
Sadly the bitfield values are not sequential (-1, -255, 0, 1,......,7, 255).  The best I could do would be to introduce an "isFlagsValid" method which would just be a switch statement to be called to verify. I don't think that's necessary though since we do all the setting of these bits